### PR TITLE
#81 remote some make styles

### LINKED ISSUE #81: Remove uses of makeStyles
If we remove the uses of `makeStyles` we can get rid of `@material-ui/styles` package (I think)

Requires changing the styles to use `styled` for:
- `LoadingApp` (Not a great name for this one?)
- `NavLink` (Also not a great name..?)
- `AppBar`
- `AppDrawer`
- `DashboardService` (Can just remove this, it's placeholder stuff)
- `RemoteDataTable`

Can also try and more flexing? 

Probably should be a couple of issues

### PR DESCRIPTION
**Pull request by joshxg**
Partial fixes #81 

- Fixes the storybook problem, which was that when I upgraded to v5 mui, the dependency of `emotion` that storybook and mui were out of sync, causing issues which made me very angry and sad and happy. This is why there are deps added for `emotion` and the resolution to a specific version. This was to my main goal which I originally thought was because of the `makeStyle` uses which I was hoping I was going to kill 3 birds with one stone (storybook/rtl/extra package). Nope
- Removed use of `makeStyle` in `AppDrawer` and `NavLink`. I ran in to a few learning curves
    - We can't pass arbitrary props to dom elements like a `div` or `a`, so when using `Styled`, you sometimes have to use `shouldForwardProps` to stop the propagation of that prop.
    - Overriding mui classes ends up being kinda nice (IMO, after figuring it out...) : https://next.material-ui.com/customization/how-to-customize/#overriding-nested-component-styles

### CODE REVIEW

File: packages/common/src/intl/IntlTestProvider.tsx [hunk 1]
```diff
@@ -21,6 +22,12 @@ export const IntlTestProvider: FC<IntlTestProviderProps> = ({
   children,
   locale,
 }) => {
+  const { locale: currentLocale, setLocale } = useHostContext();
+
+  useEffect(() => {
+    if (currentLocale !== locale) setLocale(locale);
+  }, [locale]);
+
```
Comment (joshxg) [inline packages/common/src/intl/IntlTestProvider.tsx:30]: Setting the state so it triggers or the rtl stuff in the story books

File: packages/common/src/intl/locales/en.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "app.admin": "Admin",
   "app.customer_invoices": "Customer Invoices",
-  "app.customer-invoice": "Customer Invoices",
+  "app.customer-invoice": "Invoices",
```
Comment (joshxg) [inline packages/common/src/intl/locales/en.json:4]: Uhh.. it looks better 😁

File: packages/common/src/ui/components/NavLink.tsx [hunk 3]
```diff
@@ -55,24 +65,25 @@ export const AppNavLink: FC<ListItemLinkProps> = props => {
...
-          <ListItemText primary={text} />
-        </ListItem>
-      </Tooltip>
-    </li>
+          <ListItemIcon sx={{ minWidth: 24 }}>{icon}</ListItemIcon>
+          {drawer.isOpen && <Box width={10} />}
+          {drawer.isOpen && <ListItemText primary={text} />}
+        </ListItemButton>
+      </StyledListItem>
+    </Tooltip>
```
Comment (joshxg) [inline packages/common/src/ui/components/NavLink.tsx:87]: hopefully this is the same!

File: packages/common/src/ui/components/NavLink.tsx [hunk 4]
```diff
@@ -55,24 +65,25 @@ export const AppNavLink: FC<ListItemLinkProps> = props => {
...
+    <Tooltip disableHoverListener={drawer.isOpen} title={text || ''}>
+      <StyledListItem isOpen={drawer.isOpen} isSelected={selected} to={to}>
+        <ListItemButton
+          sx={{
+            ...getListItemCommonStyles(drawer.isOpen),
+            '&.MuiListItemButton-root:hover': {
+              backgroundColor: 'transparent',
+            },
+          }}
+          disableGutters
```
Comment (joshxg) [inline packages/common/src/ui/components/NavLink.tsx:79]: this was a bit of a nightmare.. using `styled` meant I was getting weird cryptic TS warnings and things. Decided to stop and just use the `sx` prop 🤷

File: packages/common/src/ui/icons/MSupplyGuy.tsx [hunk 5]
```diff
@@ -1,8 +1,29 @@
...
-  <SvgIcon {...props} viewBox="3 1 206 310">
+const spin = keyframes`
+  from {transform:rotate(0deg);}
+  to {transform:rotate(360deg);}
+}`;
+
+const otherSpin = keyframes`
+  from {transform:rotate(360deg);}
+  to {transform:rotate(0deg);}
+}`;
```
Comment (joshxg) [inline packages/common/src/ui/icons/MSupplyGuy.tsx:14]: ignore this

File: packages/host/src/AppDrawer.stories.tsx [hunk 6]
```diff
@@ -32,3 +31,8 @@ export const Portuguese = Template.bind({});
 Portuguese.args = {
   locale: 'pt',
 };
+
+export const Arabic = Template.bind({});
+Arabic.args = {
+  locale: 'ar',
+};
```
Comment (joshxg) [inline packages/host/src/AppDrawer.stories.tsx:38]: the rtl in the storybook isn't working in that the drawer isn't going to the right, but everything else is switching up

File: packages/host/src/AppDrawer.test.tsx [hunk 7]
```diff
@@ -8,78 +9,83 @@ import {
 import { act } from 'react-dom/test-utils';
 
 describe('AppDrawer', () => {
-  it('Collapses when clicking the drawer open/close button for the first time on a large screen', () => {
-    render(
+  it('Collapses when clicking the drawer open/close button for the first time on a large screen', async () => {
+    const { getByRole, getByTestId } = render(
```
Comment (joshxg) [inline packages/host/src/AppDrawer.test.tsx:13]: these tests started failing - the `open the menu` text was changed to `close the menu` sometimes which makes sense. Also apparently `userEvent` is better to use 🤷

File: packages/host/src/AppDrawer.tsx [hunk 8]
```diff
@@ -165,37 +113,77 @@ const AppDrawer: React.FC = () => {
...
+            text={t('app.admin')}
+          />
+          <AppNavLink
+            to={AppRoute.Logout}
+            icon={<Power />}
+            text={t('app.logout')}
+          />
+        </List>
+      </ListContainer>
+    </StyledDrawer>
```
Comment (joshxg) [inline packages/host/src/AppDrawer.tsx:186]: Hopefully the same, from what I can tell!

File: packages/common/src/intl/locales/en.json (hunk 2 above)
Comment (mark-prins) [reply packages/common/src/intl/locales/en.json:4]: I've also notice that there's a `-` and a `_` 🙄  oops.
And there's a plan to rename these to `Outgoing shipment` and `Incoming shipment` 😬

**Comment by mark-prins** [comment]
so far, it's awesome. haven't quite finished going through all of it yet, already learnt a bunch thanks.
the bits I was going to dig into: 
- switching to Arabic threw an error (Cannot read property 'then' of undefined)
- the styles need a small tweak

I was going to look tomorrow, but you might get to it first which is why I'm noting it down now.

button hover colour has changed to a rich burgundy:

Screen Shot 2021-09-02 at 5 48 31 PM

left nav: when expanded the background colour and maybe the icon size
**new**:
Screen Shot 2021-09-02 at 5 46 31 PM

**zeplin**:
Screen Shot 2021-09-02 at 5 46 58 PM

when collapsed it was the circle size relative to the icons which looks odd. maybe the icon size and the button size?
**new**
Screen Shot 2021-09-02 at 5 47 17 PM
**zeplin**
Screen Shot 2021-09-02 at 5 47 30 PM

**Review by mark-prins — APPROVED** [review]
awesome!

**Comment by joshxg** [comment]
ah! Thanks so much that's much better!!

I agree re lots of things. Will be helpful in.. a year..? 😢

**Pull request merged**